operator/duties/controller.go

```diff
@@ -143,17 +184,108 @@ func createDutyExecuteMsg(duty *spectypes.Duty, pubKey *bls.PublicKey) (*spectyp
 	}, nil
 }
 
+// HandleHeadEvent handles the "head" events from the beacon node.
+func (dc *dutyController) HandleHeadEvent(event *eth2apiv1.Event) {
+	if event.Data == nil {
+		return
+	}
+
+	var zeroRoot phase0.Root
+
+	data := event.Data.(*eth2apiv1.HeadEvent)
+	if data.Slot != dc.ethNetwork.EstimatedCurrentSlot() {
+		return
+	}
+
+	// check for reorg
+	epoch := dc.ethNetwork.EstimatedEpochAtSlot(data.Slot)
+	if dc.lastBlockEpoch != 0 && epoch <= dc.lastBlockEpoch {
+		if !bytes.Equal(dc.currentDutyDependentRoot[:], zeroRoot[:]) &&
+			!bytes.Equal(dc.currentDutyDependentRoot[:], data.CurrentDutyDependentRoot[:]) {
+			dc.logger.Debug("Current duty dependent root has changed",
+				zap.String("old_dependent_root", fmt.Sprintf("%#x", dc.currentDutyDependentRoot[:])),
+				zap.String("new_dependent_root", fmt.Sprintf("%#x", data.CurrentDutyDependentRoot[:])))
+			dc.handleCurrentDependentRootChanged()
+		}
+	}
+
+	dc.lastBlockEpoch = epoch
+	dc.currentDutyDependentRoot = data.CurrentDutyDependentRoot
+}
+
 // listenToTicker loop over the given slot channel
 func (dc *dutyController) listenToTicker(slots <-chan phase0.Slot) {
 	for currentSlot := range slots {
-		// execute duties
+
+		if dc.ethNetwork.IsFirstSlotOfEpoch(currentSlot) {
+			currentEpoch := dc.ethNetwork.EstimatedEpochAtSlot(currentSlot)
+			indices := dc.validatorController.GetValidatorsIndices()
```

Comment:
the get indices might block a bit the execute duties, WDYT?

Reply:
we are using map, should be ok